datamap/lucene/src/main/java/org/apache/carbondata/datamap/lucene/LuceneDataMapFactoryBase.java

```diff
@@ -168,7 +169,8 @@ public DataMapBuilder createBuilder(Segment segment, String shardName) {
         getAllIndexDirs(tableIdentifier.getTablePath(), segment.getSegmentNo());
     for (CarbonFile indexDir : indexDirs) {
       // Filter out the tasks which are filtered through CG datamap.
-      if (!segment.getFilteredIndexShardNames().contains(indexDir.getName())) {
+      if (getDataMapLevel() != DataMapLevel.FG &&
```

Comment:
What does this for?
If it is only for CG datamap, then you can judge outside this loop.

Reply:
The design of continue is for CG. FG can't skip it by the continue, otherwise it will led to error later.